improved the dev docs (#1824)
accuknox/KubeArmor: contribution/testing_guide.md

# Testing Guide

There are two ways to check the functionalities of KubeArmor: 1) testing KubeArmor manually and 2) using the testing framework.
# 0. Make sure Kubernetes cluster is running 

> Although there are many ways to run a Kubernetes cluster (like minikube or kind) but it will not work with locally developed KubeArmor. KubeArmor needs to be on the same node as where the kubernetes nodes exist. If you will try to do this it will not identify your node since minikube and kind use virtualised node. You would either need to build your images and deploy them into these clusters or simply use `k3s` or `kubeadm`. if you are new to these terms then the most easiest way to do this is by following this guide: [K3s installation guide](k3s/README.md)

Reviewer: ```suggestion
Although there are many ways to run a Kubernetes cluster (like minikube or kind), it will not work with locally developed KubeArmor. KubeArmor needs to be on the same node as where the Kubernetes nodes exist. If you try to do this it will not identify your node since minikube and kind use virtualized nodes. You would either need to build your images and deploy them into these clusters or you can simply use `k3s` or `kubeadm` for development purposes. If you are new to these terms then the easiest way to do this is by following this guide: [K3s installation guide](k3s/README.md)
```

```text

> ~/KubeArmor/KubeArmor$ make run

Reviewer: please add `kubectl proxy` is also required while using make run

# Testing Guide

There are two ways to check the functionalities of KubeArmor: 1) testing KubeArmor manually and 2) using the testing framework.
# 0. Make sure Kubernetes cluster is running 

Although there are many ways to run a Kubernetes cluster (like minikube or kind) but it will not work with locally developed KubeArmor. KubeArmor needs to be on the same node as where the kubernetes nodes exist. If you will try to do this it will not identify your node since minikube and kind use virtualised node. You would either need to build your images and deploy them into these clusters or simply use `k3s` or `kubeadm`. if you are new to these terms then the most easiest way to do this is by following this guide: [K3s installation guide](k3s/README.md)
```text

> $ cd KubeArmor/KubeArmor
> ~/KubeArmor/KubeArmor$ make clean && make
> ~/KubeArmor/KubeArmor$ sudo -E ./kubearmor -gRPC=[gRPC port number]
>                                            -logPath=[log file path]
>                                            -enableKubeArmorPolicy=[true|false]
>                                            -enableKubeArmorHostPolicy=[true|false]
> ~/KubeArmor/tests$ make run

Reviewer 2: ```suggestion
~/KubeArmor/KubeArmor$ make run
```

make run automatically handles both the steps.

Noting we are talking about running KubeArmor here not tests. So please change this under Testing KubeArmor 1.2 and 1.3 not here

Reviewer 2: Sorry for the confusing review.

# Testing Guide

There are two ways to check the functionalities of KubeArmor: 1) testing KubeArmor manually and 2) using the testing framework.
# 0. Make sure Kubernetes cluster is running 

Although there are many ways to run a Kubernetes cluster (like minikube or kind) but it will not work with locally developed KubeArmor. KubeArmor needs to be on the same node as where the kubernetes nodes exist. If you will try to do this it will not identify your node since minikube and kind use virtualised node. You would either need to build your images and deploy them into these clusters or simply use `k3s` or `kubeadm`. if you are new to these terms then the most easiest way to do this is by following this guide: [K3s installation guide](k3s/README.md)

> # 1.  Test KubeArmor manually

Reviewer 2: Under this can we merge 1.2 and 1.3 and directly ask them to run
```
make run 
```

> Although there are many ways to run a Kubernetes cluster (like minikube or kind) but it will not work with KubeArmor. If you will try to do this it will not identify your node. So use `k3s` or `kubeadm`. if you are new to these terms then the most easiest way to do this is by following this guide: [K3s installation guide](k3s/README.md)

Reviewer 2: ```suggestion
Although there are many ways to run a Kubernetes cluster (like minikube or kind) but it will not work with locally developed KubeArmor. KubeArmor needs to be on the same node as where the kubernetes nodes exist. If you will try to do this it will not identify your node since minikube and kind use virtualised node. You would either need to build your images and deploy them into these clusters or simply use `k3s` or `kubeadm`. if you are new to these terms then the most easiest way to do this is by following this guide: [K3s installation guide](k3s/README.md)
```

    ```text
    ~/KubeArmor/tests$ cat /tmp/kubearmor.test
    ```

> - To run a specific suit of tests move to the directoory of test and run

Reviewer: ```suggestion
- To run a specific suit of tests move to the directory of test and run
```
